lib/cmd/fh3/appdata/export/status.js

```diff
@@ -0,0 +1,123 @@
+"use strict";
+
+var fs = require('fs')
+  , _ = require('underscore')
+  , fhreq = require('../../../../utils/request.js')
+  , async = require('async')
+  , ProgressBar = require('progress')
+  , common = require('../../../../common');
+
+var POLL_INTERVAL = 2000;
+var MIN_INTERVAL = 100;
+var EXIT_KEYS = ['q', 'Q'];
+
+/**
+ * Read progress information by calling the `read` endpoint to get the
+ * job definition. Progress information is included there:
+ * - step: current export job step
+ * - totalSteps: total export steps
+ */
+function readProgress(params, cb) {
+  var url = 'api/v2/appdata/' + params.envId + '/' + params.appId + '/export/data/' + params.jobId;
+  common.doGetApiCall(fhreq.getFeedHenryUrl(), url, 'Error reading progress', cb);
+}
+
+/**
+ * Update the progressbar if possible. Don't use `bar#tick` as that will not
+ * set the current progress but add the new valud to the current progress.
+ */
+function printout(data, bar) {
+  if (data.totalSteps && data.step >= 0) {
+    bar.update(data.step / data.totalSteps);
+  } else {
+    /*ignore jslint start*/
+    console.log('No progress information in response');
+    /*ignore jslint end*/
+  }
+}
+
+/**
+ * Get a sane poll interval (use the default one if the user did not provide
+ * one)
+ */
+function getPollInterval(params) {
+  if (params.interval && _.isNumber(params.interval)) {
+    var interval = params.interval;
+    return interval >= MIN_INTERVAL ? interval : POLL_INTERVAL;
+  } else {
+    return POLL_INTERVAL;
+  }
+}
+
+module.exports = {
+  'desc': 'Poll for job status',
+  'examples': [{
+    cmd: [
+      'fhc addpata export status --appId=5ujx2eifvzaudq43nw4nmvcu --envId=live --jobId=5731da1dd3e2b283203c4054'
+    ].join(''),
+    desc: 'Poll for job status'
+  }],
+  'demand': ['envId', 'appId', 'jobId'],
+  'alias': {},
+  'describe': {
+    'envId': 'Environment id',
+    'appId': 'Application id',
+    'jobId': 'Export job id',
+    'interval': 'Polling interval in milliseconds (default is 2000)'
+  },
+  'method': 'get',
+  'customCmd': function (params, cb) {
+    /*ignore jslint start*/
+    // Need to use `console#log` here until we introduce some other logging or outpur
+    // framework to fhc
+    console.log('Polling job progress. Press `q` to quit');
+    /*ignore jslint end*/
+
+    function onInterval() {
+      readProgress(params, function (err, result) {
+        if (err) {
+          clearInterval(poller);
+          return cb(err);
+        }
+
+        // supercore sends the actual result in the `data` field
+        result = result.data;
+
+        if (bar == null && result.totalSteps) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the declaration of "bar" before this usage. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage)